motor/core.py

```diff
@@ -810,7 +836,7 @@ def __getitem__(self, name):
         )
 
         return collection_class(
-            self.database, self.name + "." + name, _delegate=self.delegate[name]
+            self.database, self.name + "." + str(name), _delegate=self.delegate[name]
```

Comment:
I don't think we want to call str(name) since name should already be a str.

Reply:
At least one of the tests was passing in an integer for the name

Reply:
We should skip those tests. Motor should not accept non-strings for getitem even if pymongo does.